### PR TITLE
feat(IPVC-2276): skip accessions from gff that are not in the tx_info file, log missing acs to file

### PR DESCRIPTION
**Pull request by sptaylor**
The exonset file, derived from gff files, contains transcript accessions that are not present in the transcript info file/Seqrepo, and this causes downstream issues. We are adding a step to filter out these missing transcripts from the exonsets file.
Demo:
```
+ sbin/filter_exonset_transcripts.py --tx-info /workdir/loading/gbff.txinfo.gz --exonsets /workdir/loading/gff.exonsets.gz --missing-ids /workdir/loading/filtered_tx_acs.txt
+ tee /workdir/logs/filter_exonset_transcripts.log
+ gzip -c
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NM_000853.3 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NR_003491.3 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NR_033319.2 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NR_033320.2 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NR_033321.2 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NR_156186.2 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NM_001284286.1 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NM_001284289.1 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NM_001284288.1 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NM_001002837.2 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NM_001284287.1 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 WARNING  [__main__] Exon set transcript NM_001350317.2 not found in txinfo file. Filtering out.
2024-04-04 20:42:22 INFO     [__main__] Filtered out exon sets for 12 transcripts: NR_033320.2,NM_001284286.1,NM_001350317.2,NR_033319.2,NR_156186.2,NM_000853.3,NM_001284287.1,NM_001284288.1,NM_001002837.2,NR_003491.3,NM_001284289.1,NR_033321.2
```
The filtered exonsets are printed to stdout and the missing transcript ids are printed to file.

### CODE REVIEW

File: sbin/ncbi_parse_genomic_gff.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def _get_exon_number_from_id(alignment_id: str) -> int:
     return int(alignment_id.split("-")[-1])
 
 
-def parse_gff_file(file_paths: List[str]) -> dict[str, List[GFFRecord]]:
+def parse_gff_files(file_paths: List[str]) -> dict[str, List[GFFRecord]]:
```
Comment (sptaylor) [inline sbin/ncbi_parse_genomic_gff.py:107]: Updated the name to reflect the function takes a list of files

File: sbin/exonset-to-seqinfo [hunk 2]
```diff
@@ -49,6 +49,8 @@ if __name__ == "__main__":
     ac_re = re.compile("[NX][CGMPR]_")
 
     opts = parse_args(sys.argv[1:])
+    input_dir = os.path.dirname(opts.FILES[0])
```
Comment (bsgiles73) [inline sbin/exonset-to-seqinfo]: Can we add an output directory to the supported arguments. So we can have this file directed to the directory the user chooses?

**Review by bsgiles73 — COMMENTED** [review]
My description in the ticket really made it look like this was an issue with `ncbi_parse_genomic_gff.py`, but it is not. Sorry for the confusion. But now I have a question. If we have extra transcripts in the GFF files, that are missing from SeqRepo, how many places might break? You definitely found one `exonset-to-seqinfo`. If the extra transcripts are in the GFF. That means they should be in the intermediate file. Which means they get into the databaes, via the `uta load-exonset` step? Or are they skipped because the transcripts are missing from the `transcript` table?

File: sbin/filter_exonset_transcripts.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+from uta.tools.file_utils import open_file
+
+logging_conf_fn = importlib_resources.files("uta").joinpath("etc/logging.conf")
+logging.config.fileConfig(logging_conf_fn)
+logging.getLogger().setLevel(logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def filter_exonset(exonset_file, transcript_ids, missing_ids_file):
+    with open_file(exonset_file) as es_f, open(missing_ids_file, 'w') as missing_f:
```
Comment (bsgiles73) [inline sbin/filter_exonset_transcripts.py:21]: Nice!

**Review by bsgiles73 — APPROVED** [review]
LGTM

**Pull request merged**